src/dss/create_notebook.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from lightkube import Client
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import (
+    DSS_CLI_MANAGER_LABELS,
+    DSS_NAMESPACE,
+    FIELD_MANAGER,
+    MANIFEST_TEMPLATES_LOCATION,
+    NOTEBOOK_PVC_NAME,
+)
+from dss.logger import setup_logger
+from dss.utils import get_mlflow_tracking_uri, get_notebook_url, wait_for_deployment_ready
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def create_notebook(name: str, image: str, lightkube_client: Client) -> None:
+    """
+    Creates a Notebook server on the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        image (str): The image used for the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+    manifests_file = Path(
+        Path(__file__).parent, MANIFEST_TEMPLATES_LOCATION, "notebook_deployment.yaml.j2"
+    )
+
+    config = _get_notebook_config(image, name)
+
+    # Initialize KubernetesResourceHandler
+    k8s_resource_handler = KubernetesResourceHandler(
+        field_manager=FIELD_MANAGER,
+        labels=DSS_CLI_MANAGER_LABELS,
+        template_files=[manifests_file],
+        context=config,
+        resource_types={Deployment, Service},
+        lightkube_client=lightkube_client,
+    )
+
+    try:
+        k8s_resource_handler.apply()
```

Comment:
Should we handle`lightkube` errors? I think that if for any reason the notebook server deployment cannot be applied, it will throw an error instead of a timeout, which is not handled.

Also, what about the possibility of a person creating a notebook server with the same name, but different image, do you think applying will override the existing server? Should we have an imperative approach here to avoid modifying existing deployments?

Reply:
Added the check in create_notebook

Reply:
> Yeah I see what you mean about the `lightkube` errors - do you have an opinion on what we should do? The only thing I can think of is catching lightkube errors (ApiError?) and maybe raising a new error from that that adds us saying that something went wrong but we don't know what? I think we'd want to expose lightkube's traceback (since there's probably useful debugging info there) but we should add some value too

We could expose the traceback in a `debug` log so it helps when issues are filed. We'll need a mechanism for exposing debug lugs, a verbose mode if you will.
Regarding the message users will get, perhaps this is a question for the UX team, but for now I think it would be nice to show at least a msg that says there was an issue, the issue code, and a suggested fix(?).

Reply:
> Since every DSS notebook uses the same PVC, I don't think there's much value in dss overwriting existing notebooks. Users can just remove and create them without losing their data. For now I'll add a check for an existing notebook of the same name, and warn the user if that happens. wdyt?

I think it is fine if we warn users for now as this is also a good question for the UX team. Let's follow up with them.

Reply:
> Since every DSS notebook uses the same PVC, I don't think there's much value in dss overwriting existing notebooks. Users can just remove and create them without losing their data. For now I'll add a check for an existing notebook of the same name, and warn the user if that happens. wdyt?

in the updated spec with the UX team's feedback:
```
Output upon name failure exists:
# fallback error
Failed to create notebook. Notebook <NAME> already exists.Please specify a different name.
```
Andrew's change is similar to that, just the error message needs a little tweaking.